### PR TITLE
test: add testing for updating resource of container

### PR DESCRIPTION
**Pull request by chuanchang**
As summary.

### CODE REVIEW

File: pkg/validate/container.go [hunk 1]
```diff
@@ -704,3 +742,16 @@ func verifyLogContents(podConfig *runtimeapi.PodSandboxConfig, logPath string, l
 	}
 	Expect(found).To(BeTrue(), "expected log %q (stream=%q) not found in logs %+v", log, stream, msgs)
 }
+
+// verifyUpdateContainerResources verifies updates resource of container.
+func verifyUpdateContainerResources(rc internalapi.RuntimeService, containerID string, Resources *runtimeapi.LinuxContainerResources, expectError bool, expectedErrStr string) {
+    out := rc.UpdateContainerResources(containerID, Resources)
+    msg := fmt.Sprintf("%v", out)
```
Comment (zhuangqh) [inline pkg/validate/container.go]: This line is weird 🧐.

File: pkg/validate/container.go [hunk 2]
```diff
@@ -144,6 +144,44 @@ var _ = framework.KubeDescribe("Container", func() {
...
+            Resources := &runtimeapi.LinuxContainerResources{
+                CpuPeriod:          1000,
+                CpuQuota:           0,
+                CpuShares:          1024,
+                CpusetCpus:         "",
+                CpusetMems:         "",
+                MemoryLimitInBytes: 4194304,
+                OomScoreAdj:        0,
+            }
+            expectError := false
```
Comment (Starnop) [inline pkg/validate/container.go]: not necessary?

File: pkg/validate/container.go [hunk 3]
```diff
@@ -144,6 +144,44 @@ var _ = framework.KubeDescribe("Container", func() {
...
+            expectedErrStr := ""
```
Comment (Starnop) [inline pkg/validate/container.go]: so do with upon.

File: pkg/validate/container.go [hunk 4]
```diff
@@ -143,6 +143,44 @@ var _ = framework.KubeDescribe("Container", func() {
...
+				CpuQuota:           0,
+				CpuShares:          1024,
+				CpusetCpus:         "",
+				CpusetMems:         "",
+				MemoryLimitInBytes: 4194304,
+				OomScoreAdj:        0,
+			}
+			expectError := false
+			expectedErrStr := ""
+			verifyUpdateContainerResources(rc, containerID, Resources, expectError, expectedErrStr)
```
Comment (zhuangqh) [inline pkg/validate/container.go]: We should check whether the value under `/sys/fs/cgroup` is meeting our expectation.

Comment (chuanchang) [reply pkg/validate/container.go]: sure, I have added cgroup related check in v2 patch.

**Comment by zhuangqh** [comment]
LGTM

**Pull request merged**